tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
```

Comment:
Some of the tests used the condition checking that you mentioned before, but the events were predictable enough that I just used the counts. If you feel we need the condition based then I'll prob refactor accordingly.